src/modules/src/controller_lee_payload.c

```diff
@@ -226,101 +204,67 @@ static controllerLeePayload_t g_self = {
 
 static void runQP(const struct QPInput *input, struct QPOutput* output)
 {
```

Comment:
some of these local variables will now have undefined values, which is a bit scary. I suggest a structure like the following:
```
// declare common variables:

if (num_neighbors >= 2) {
   // declare additional variables
   
   if (num_neighbors >= 3) {
     // declare additional variables
   }
}
```

Not the most pretty if it goes very deep, but functional and avoids using uninitialized memory.

Reply:
Nested ifs, got it!